src/olympia/addons/serializers.py

```diff
@@ -415,6 +416,16 @@ def get_url(self, obj):
         # annoying in subclasses which don't want to load that version.
         return absolutify(obj.get_detail_url())
 
+    def get_contributions_url(self, obj):
+        if not obj.contributions:
+            # don't add anything when it's not set.
+            return obj.contributions
+        parts = urlsplit(obj.contributions)
+        query = '&'.join(
+            s for s in (parts.query, amo.CONTRIBUTE_UTM_PARAMS) if s)
```

Comment:
I would make `CONTRIBUTE_UTM_PARAMS` a `dict` and I would parse `parts.query` with `parse_qs` (probably), then merge both dicts and finally use `urlencode`. That would be arguably more readable that this line and that would allow us to override the `utm_*` query params if the `contributions` URL has query params including one of the UTM params we want to set.

Reply:
QueryDicts explicitly _do_ allow duplicate keys though https://docs.djangoproject.com/en/2.2/ref/request-response/#querydict-objects.  Do we want to override a user's UTM params with our own anyway? - that seems like a bad thing given that we're adding these to all services not just donate.m.o (and wasn't specified in the issue)

Reply:
ha, right 🤦‍♂ Then, I guess we can keep duplicate keys then and use `QueryDict.update()`, that would still be slightly more readable in my opinion 😛

Reply:
https://github.com/mozilla/addons-server/pull/13042/commits/6da0d9db2fb30601a126e1188fba8957c85b5aeb
Ironically, I started the original patch using `parse_qs` and then changed to a flat string because I thought it was overwritten and made it less readable.